ocaml/xapi/xapi_host.ml

```diff
@@ -792,18 +792,12 @@ let management_reconfigure ~__context ~pif =
   if Db.Pool.get_ha_enabled ~__context ~self:pool then
     raise (Api_errors.Server_error(Api_errors.ha_is_enabled, []));
 
-  (* Plugging a bond slave is not allowed *)
-  if Db.PIF.get_bond_slave_of ~__context ~self:pif <> Ref.null then
```

Comment:
Oops: we need to do this check even if the pif is not managed, i.e. it must not depend on the `if Db.PIF.get_managed` below.

So I think we might need to make the call to shared test from around here something like
`Xapi_pif.assert_can_plug ~__context ~for_mgmt:false ~required_address_type:None ~self:pif`

whereas in the Pool function it would be
`Xapi_pif.assert_can_plug ~__context ~for_mgmt:true ~required_address_type:(Some address_type) ~self`
(where `address_type` is the type from the first PIF in the list, as at present).

So then the implementation would be something like
```
let assert_can_plug ~__context ~for_mgmt ~required_address_type ~self =
  assert_not_a_bond_slave ~__context ~self;
  let primary_address_type = Db.PIF.get_primary_address_type ~__context ~self in
  if for_mgmt and required_address_type <> (Some primary_address_type) then
    raise (Api_errors.Server_error(Api_errors.pif_incompatible_primary_address_type, [Ref.string_of self]));
  if Db.PIF.get_managed ~__context ~self:pif = true then begin
    match primary_address_type with
      | `IPv4 -> assert_valid_ip_configuration ~__context ~self 
      | `IPv6 -> assert_valid_ipv6_configuration ~__context ~self 
      | _ -> try
        let mgmt_pif = get_management_interface ~__context ~host:(Helpers.get_localhost ~__context) in
        let mgmt_address_type = Db.PIF.get_primary_address_type ~__context ~self:mgmt_pif in
        if primary_address_type <> mgmt_address_type then
          raise (Api_errors.Server_error(Api_errors.pif_incompatible_primary_address_type, []));
        with _ ->
        (); (* no current management interface *)
```